packages/store/src/patterns/patternMatchers.js

```diff
@@ -1317,133 +1332,226 @@ const makePatternKit = () => {
+            X`Expected at least ${q(
+              requiredPatt.length,
+            )} arguments: ${specimen}`,
+          )) &&
+        requiredPatt.every((p, i) =>
+          // eslint-disable-next-line no-plusplus
+          checkMatches(requiredSpecimen[i], p, check, `arg ${argNum++}`),
+        ) &&
+        partialPatt.every((p, i) =>
+          // eslint-disable-next-line no-plusplus
+          checkMatches(optionalSpecimen[i], p, check, `arg ${argNum++}?`),
+        ) &&
+        checkMatches(restSpecimen, restPatt, check, '...rest')
       );
     },
 
-    checkIsWellFormed: (splitArgs, check) => {
+    checkIsWellFormed: (splitArray, check) => {
       if (
-        passStyleOf(splitArgs) === 'copyArray' &&
-        (splitArgs.length === 1 || splitArgs.length === 2)
+        passStyleOf(splitArray) === 'copyArray' &&
+        (splitArray.length >= 1 || splitArray.length <= 3)
       ) {
-        const [base, rest = undefined] = splitArgs;
-        const baseStyle = passStyleOf(base);
+        const [requiredPatt, optionalPatt = undefined, restPatt = undefined] =
+          splitArray;
         if (
-          isPattern(base) &&
-          (baseStyle === 'copyArray' || baseStyle === 'copyRecord') &&
-          (rest === undefined || isPattern(rest))
+          isPattern(requiredPatt) &&
+          passStyleOf(requiredPatt) === 'copyArray' &&
+          (optionalPatt === undefined ||
+            (isPattern(optionalPatt) &&
+              passStyleOf(optionalPatt) === 'copyArray')) &&
+          (restPatt === undefined || isPattern(restPatt))
         ) {
           return true;
         }
       }
       return check(
         false,
-        X`Must be an array of a base structure and an optional rest pattern: ${splitArgs}`,
+        X`Must be an array of a requiredPatt array, an optional optionalPatt array, and an optional restPatt: ${q(
+          splitArray,
+        )}`,
       );
     },
 
-    getRankCover: ([base, _rest = undefined]) =>
-      getPassStyleCover(passStyleOf(base)),
+    getRankCover: ([
+      _requiredPatt,
+      _optionalPatt = undefined,
+      _restPatt = undefined,
+    ]) => getPassStyleCover('copyArray'),
 
-    checkKeyPattern: ([base, _rest = undefined], check) =>
-      check(false, X`${q(passStyleOf(base))} not yet supported as keys`),
+    checkKeyPattern: (
+      [_requiredPatt, _optionalPatt = undefined, _restPatt = undefined],
+      check,
+    ) => check(false, X`copyRecord not yet supported as keys`),
   });
 
+  /**
+   * @param {CopyRecord<Passable>} specimen
+   * @param {CopyRecord<Pattern>} requiredPatt
+   * @param {CopyRecord<Pattern>} optionalPatt
+   * @returns {{
+   *   requiredSpecimen: CopyRecord<Passable>,
+   *   optionalSpecimen: CopyRecord<Passable>,
+   *   restSpecimen: CopyRecord<Passable>
+   * }}
+   */
+  const splitRecordParts = (specimen, requiredPatt, optionalPatt) => {
+    // Not frozen! Mutated in place
+    /** @type {[string, Passable][]} */
```

Comment:
nit, this style is little clearer for arrays of tuples:
```suggestion
    /** @type {Array<[string, Passable]>} */
```

Reply:
Not sure this is good to do as a one-off in this PR. Between agoric-sdk and endo, we currently have 45 occurrences of "][]" which, by an informal skim, seem to be all or almost all types. OTOH we have 21 occurrences of "Array<[".

If we decide to canonicalize this element of style, we should coordinate across all the occurrences we wish to change.

Reply:
Thanks for looking. To be clear, I wasn't requesting this change. By "nit" I meant to just suggest. (22:45 instead of 21:46). I've had a lint rule before to enforce the `Array<>` for tuples. Since we're on the topic, would you support such a rule?

Reply:
What does "22:45" or "21:46" mean?

We have not discussed, so this opinion is offered in ignorance of contrary arguments: I would not support. For arrays and tuples, I prefer the square bracket notation. I find it clearer as well as more concise.

Reply:
> What does "22:45" or "21:46" mean?

Shorthand for ratios of instances of the two styles. (You said the current ratio is 21:45 so in this thread we've been talking about which one to increment.)